prow/pjutil/jobspec.go

```diff
@@ -44,23 +48,51 @@ type JobSpec struct {
 	agent kube.ProwJobAgent
 }
 
-func NewJobSpec(spec kube.ProwJobSpec, buildId string) JobSpec {
+func NewJobSpec(spec kube.ProwJobSpec, buildId, prowJobId string) JobSpec {
 	return JobSpec{
-		Type:    spec.Type,
-		Job:     spec.Job,
-		BuildId: buildId,
-		Refs:    spec.Refs,
-		agent:   spec.Agent,
+		Type:      spec.Type,
+		Job:       spec.Job,
+		BuildId:   buildId,
+		ProwJobId: prowJobId,
+		Refs:      spec.Refs,
+		agent:     spec.Agent,
 	}
 }
 
+// GetBuildID calls out to `tot` in order
+// to vend build identifier for the job
+func GetBuildID(name, totURL string) (string, error) {
+	var err error
+	url := totURL + "/vend/" + name
+	for retries := 0; retries < 60; retries++ {
+		if retries > 0 {
+			time.Sleep(2 * time.Second)
+		}
+		var resp *http.Response
+		resp, err = http.Get(url)
+		if err != nil {
+			continue
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != 200 {
+			continue
+		}
+		if buf, err := ioutil.ReadAll(resp.Body); err == nil {
+			return string(buf), nil
+		}
+		return "", err
```

Comment:
If `ReadAll` fails we won't return an error here because `err` is redefined instead of just being assigned in the conditional.